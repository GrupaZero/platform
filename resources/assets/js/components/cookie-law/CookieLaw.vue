<style lang="scss">
  .cookies-info {
    position        : fixed;
    overflow        : hidden;
    box-sizing      : border-box;
    z-index         : 9999;
    width           : 100%;
    display         : flex;
    justify-content : space-between;
    align-items     : baseline;
    bottom          : 0;
    background      : rgba(255, 255, 255, 0.9);
    padding         : 10px;
    box-shadow      : -1px 3px 8px #000;
  }
</style>

<template>
  <transition appear name="slideFromBottom">
    <div class="cookies-info container" v-if="isOpen">
      <div class="cookies-info-message">
        {{ $t('cookieLaw.message') }}
        <a :href="policyUrl">{{ $t('cookieLaw.link_text') }}</a>
      </div>
      <button class="btn btn-success btn-sm" @click="accept">{{ $t('cookieLaw.button_text') }}</button>
    </div>
  </transition>
</template>

<script>
    import Cookies from 'js-cookie'
    export default {
        props: {
            policyUrl: {
                type: String,
                required: true
            }
        },
        data () {
            return {
                isOpen: false
            }
        },
        created () {
            if (!this.getVisited() === true) {
                this.isOpen = true
            }
        },
        methods: {
            _getCookiesInstance () {
                return Cookies
            },
            setVisited () {
                this._getCookiesInstance().set('cookies_policy', 'accepted', {expires: 365})
            },
            getVisited () {
                console.log(this);
                return this._getCookiesInstance().get('cookies_policy')
            },
            accept () {
                this.setVisited()
                this.isOpen = false
            }
        }
    }
</script>