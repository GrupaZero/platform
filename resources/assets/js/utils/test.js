import Translations from '../lang/locales.js';
import VueI18n from 'vue-i18n';
import {mount} from 'avoriaz';

function vueMount(component, arg) {
    component.i18n = new VueI18n({
        locale: 'en',
        fallbackLocale: 'en',
        messages: Translations
    });
    return mount(component, arg)
}

export {vueMount};