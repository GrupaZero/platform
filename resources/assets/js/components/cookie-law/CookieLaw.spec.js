import test from 'ava'
import sinion from 'sinon'
import {vueMount} from '../../utils/test'
import Bluebird from 'bluebird'
import CookieLaw from './CookieLaw.vue'

test.serial('Should display cookies law info', t => {
    t.plan(1)

    CookieLaw.methods._getCookiesInstance = {
        get: sinion.stub(),
        set: sinion.stub()
    }


    const wrapper = vueMount(CookieLaw, {
        propsData: { policyUrl: 'xxx' }
    })

    return Bluebird
    .delay()
    .then(() => {
        t.snapshot(wrapper.html())
    })

})