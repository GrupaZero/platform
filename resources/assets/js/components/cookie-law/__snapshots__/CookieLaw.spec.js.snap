// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should display cookies law info 1`] = `
"<div class=\\"cookies-info container slideFromBottom-enter slideFromBottom-enter-active\\"><div class=\\"cookies-info-message\\">
        We use cookies and similar technologies. By using the Site you agree to the use of cookies, 
        <a href=\\"xxx\\">read more about our cookies policy.</a></div><button class=\\"btn btn-success btn-sm\\">Ok</button></div>"
`;
